fidesctl/src/fidesapi/crud.py

```diff
@@ -15,13 +15,28 @@
 from fideslang import model_map
 
 
+class NotFoundError(HTTPException):
+    """
+    To be raised when a requested resource does not exist in the database.
+    """
+
+    def __init__(self, resource_type: str, fides_key: str) -> None:
+        detail = {
+            "error": "resource does not exist",
+            "resource_type": resource_type,
+            "fides_key": fides_key,
+        }
+
+        super().__init__(status.HTTP_422_UNPROCESSABLE_ENTITY, detail=detail)
```

Comment:
I think a 422 is supposed to be for malformed data specifically, and it is the default error that FastAPI returns when a request comes in and is invalid.

For a true `NotFound`, I think 404 is a better code

Reply:
I originally had a 404 response here, but I was convinced to switch to a 422 after re-reading [the spec](https://datatracker.ietf.org/doc/html/rfc4918#section-11.2). It implies to me that because the requester hit a valid route (but included a bad request payload), a 404 would be inappropriate.

Reply:
the same spec mentions this though: https://datatracker.ietf.org/doc/html/rfc4918#section-9.6, i.e. after a delete a get of the same resource should throw a 404

It isn't that the request is malformed or had bad data, the resource itself just doesn't exist in the database. 404 can be used for both "this route doesn't exist" and also "this resource doesn't exist"

Reply:
Ok, yea I don't feel strongly about returning a 422 for this. I changed it to 404.